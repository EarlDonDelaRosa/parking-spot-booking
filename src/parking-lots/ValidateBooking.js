import React, { useState } from "react";
import { Formik } from 'formik';
import * as Yup from 'yup';
import UserPage from "./UserPage";
import { Button, Modal } from "react-bootstrap";
import axios from "axios";
import useForm from "./UseForm";

function ValidateBooking() {
    const URL = 'http://localhost:3000/';
    const [showModal, setShowModal] = useState(false);
    const [bookDate, setBookDate] = useState({});
    const { bookings, setBookings } = useForm();

    // Errors for incorrect input --by earl
    const ErrorSchema = Yup.object().shape({
        date: Yup.date()
            .min(new Date(), 'No more slots for today')
            .required('Date is required'),
        time: Yup.string()
            .required('Time is required'),
        slot: Yup.string()
            .required('Slot is required')
    })

    // Modal to show --by earl
    const handleBook = value => {
        setBookDate(value);
        setShowModal(true);
    }

    const closeModal = () => {
        setShowModal(false);
    }

    const handleModalConfirm = () => {
        let userId = localStorage.getItem('userId');
        axios.post(URL + 'bookings/', {
            ...bookDate,
            status: 'Pending',
            userId: "id"+userId,
            keyid: bookings.length + 1,
        })
        .then((res) => console.log(res))

        axios.post(`${URL}id${userId}/`, {
            ...bookDate,
            status: 'Pending'
        })
        .then((res) => console.log(res))
        setShowModal(false);
        // document.location.reload()
    }

    // Formik validation --by earl
    return (
        <div>
            <Formik 
                initialValues={{
                    date: '',
                    time: '',
                    slot: ''
                }}
                onSubmit={(value) => {
                    handleBook(value)
                    console.log(value)
                }} 
                validationSchema={ErrorSchema}
                component={ UserPage }
            />

                {/* Modal for booking confirmation */}
                <Modal
                    show={showModal}
                    onHide={closeModal}
                    backdrop="static"
                    aria-labelledby="contained-modal-title-vcenter"
                    centered
                    keyboard={false}
                >
                <Modal.Header closeButton>
                    <Modal.Title>Confirmation</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    <div>
                        Request a parking spot on { bookDate.date }<br/>
                        with { bookDate.slot } slots for { bookDate.time }?<br/>
                        Proceed?
                    </div>
                </Modal.Body>
                <Modal.Footer>
                    <Button variant="secondary" onClick={closeModal}>Cancel</Button>
                    <Button variant="primary" onClick={handleModalConfirm}>Confirm</Button>
                </Modal.Footer>
                </Modal>
                {/* Modal */}
        </div>
    )
}

export default ValidateBooking