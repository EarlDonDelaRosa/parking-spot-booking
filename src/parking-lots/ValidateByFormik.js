import React, { useContext } from "react";
import LoginPage from "./LoginPage";
import { Formik } from 'formik';
import * as Yup from 'yup';
import useURL from "./useURL";
import UserContext from "./UserContext";

function ValidateByFormik() {
    const URL = 'http://localhost:3000/userlist/';
    const USERLIST = useURL(URL);
    // const { setUser } = useContext(UserContext);

    // Errors for incorrect input --by earl
    const ErrorSchema = Yup.object().shape({
        username: Yup.string()
            .required('Username is required')
            .min(2, 'Username is too short')
            .max(15, 'Username is too long'),
        password: Yup.string()
            .required('Password is required')
    })

    // Validation to match database --by earl
    const handleLogin = value => {
        for (let i = 0; i < USERLIST.length; i++) {
            if (USERLIST[i].username === value.username) {
                if (USERLIST[i].password === value.password) {
                    // alert(JSON.stringify(value));
                    // setUser(USERLIST[i].name);
                    if (USERLIST[i].admin === 'Y') {
                        window.location.replace('/book-a-parking-spot/admin');
                        break;
                    } else {
                        localStorage.setItem('userId', USERLIST[i].id);
                        window.location.replace('/book-a-parking-spot/userpage');
                        break;
                    }
                }
                alert('Password did not matched');
                break;
            }
        }
    }

    // Formik validation --by earl
    return (
        <div>
            <Formik 
                initialValues={{
                    username: '',
                    password: ''
                }}
                onSubmit={(value) => {
                    handleLogin(value)
                }} 
                validationSchema={ErrorSchema}
                component={ LoginPage }/>
        </div>
    )
}

export default ValidateByFormik