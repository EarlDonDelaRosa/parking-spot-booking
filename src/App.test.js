import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import ValidateBooking from './parking-lots/ValidateBooking';
import ValidateByFormik from './parking-lots/ValidateByFormik';

describe('This is login page', () => {
  beforeEach(() => {
    render(<ValidateByFormik />);
  })

  test('h2 Login', () => {
    const linkElement = screen.getByText(/Login/);
    expect(linkElement).toBeInTheDocument();
  });

  test('To check if data has changed', () => {
    const linkElement = screen.getByText(/Login/);
    expect(linkElement).toMatchSnapshot();
  });

  test('placeholder for Username test', () => {
    const linkElement = screen.getByPlaceholderText(/Username/);
    expect(linkElement).toBeInTheDocument();
  })

  test('placeholder for Username test', () => {
    const linkElement = screen.getByPlaceholderText(/Password/);
    expect(linkElement).toBeInTheDocument();
  })

  // test('button check', () => {
  //   const linkElement = screen.getAllByRole('button');
  //   fireEvent.click(linkElement);
  //   expect(linkElement).toHaveLength(1);
  // })

  test('input label', () => {
    const linkElement = screen.getByText(/Username/);
    expect(linkElement).toBeInTheDocument();
  })
})

describe('This is login page', () => {
  beforeEach(() => {
    render(<ValidateBooking />, {wrapper: Router});
  })

  test('h5 Welcome', () => {
    const linkElement = screen.getByText(/Welcome/);
    expect(linkElement).toBeInTheDocument();
  });

  test('To check if data has changed', () => {
    const linkElement = screen.getByText(/Welcome/);
    expect(linkElement).toMatchSnapshot();
  });
})