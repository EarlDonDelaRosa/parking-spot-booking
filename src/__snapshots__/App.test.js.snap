// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is login page To check if data has changed 1`] = `
<h2>
  Login
</h2>
`;

exports[`This is login page To check if data has changed 2`] = `
<h5>
  Welcome 
</h5>
`;
